dGame/dBehaviors/SpeedBehavior.cpp

```diff
@@ -7,85 +7,47 @@
 
 
 void SpeedBehavior::Handle(BehaviorContext* context, RakNet::BitStream* bitStream, BehaviorBranchContext branch) {
-	if (m_AffectsCaster) {
-		branch.target = context->caster;
-	}
+	if (m_AffectsCaster) branch.target = context->caster;
 
 	auto* target = EntityManager::Instance()->GetEntity(branch.target);
-
-	if (target == nullptr) {
-		return;
-	}
+	if (!target) return;
 
 	auto* controllablePhysicsComponent = target->GetComponent<ControllablePhysicsComponent>();
-
-	if (controllablePhysicsComponent == nullptr) {
-		return;
-	}
+	if (!controllablePhysicsComponent) return;
 
 	const auto current = controllablePhysicsComponent->GetSpeedMultiplier();
-
 	controllablePhysicsComponent->SetSpeedMultiplier(current + ((m_RunSpeed - 500.0f) / 500.0f));
-
 	EntityManager::Instance()->SerializeEntity(target);
 
 	if (branch.duration > 0.0f) {
 		context->RegisterTimerBehavior(this, branch);
 	} else if (branch.start > 0) {
-		controllablePhysicsComponent->SetIgnoreMultipliers(true);
-
 		context->RegisterEndBehavior(this, branch);
 	}
 }
 
-void SpeedBehavior::Timer(BehaviorContext* context, BehaviorBranchContext branch, LWOOBJID second) {
-	auto* target = EntityManager::Instance()->GetEntity(branch.target);
-
-	if (target == nullptr) {
-		return;
-	}
-
-	auto* controllablePhysicsComponent = target->GetComponent<ControllablePhysicsComponent>();
-
-	if (controllablePhysicsComponent == nullptr) {
-		return;
-	}
-
-	const auto current = controllablePhysicsComponent->GetSpeedMultiplier();
-
-	controllablePhysicsComponent->SetSpeedMultiplier(current - ((m_RunSpeed - 500.0f) / 500.0f));
-
-	EntityManager::Instance()->SerializeEntity(target);
+void SpeedBehavior::Calculate(BehaviorContext* context, RakNet::BitStream* bitStream, BehaviorBranchContext branch) {
+	Handle(context, bitStream, branch);
 }
 
 void SpeedBehavior::End(BehaviorContext* context, BehaviorBranchContext branch, LWOOBJID second) {
 	auto* target = EntityManager::Instance()->GetEntity(branch.target);
-
-	if (target == nullptr) {
-		return;
-	}
+	if (!target) return;
 
 	auto* controllablePhysicsComponent = target->GetComponent<ControllablePhysicsComponent>();
-
-	if (controllablePhysicsComponent == nullptr) {
-		return;
-	}
+	if (!controllablePhysicsComponent) return;
 
 	const auto current = controllablePhysicsComponent->GetSpeedMultiplier();
-
 	controllablePhysicsComponent->SetIgnoreMultipliers(false);
```

Comment:
This line is not done in the Timer function, have you made sure this doesn't cause any issues setting this when it was not in the past?

Reply:
It's just a fail safe to enable the update in serialization, probably not needed in the first place, but it was there. This is the only place in the codebase that `SetIgnoreMultipliers` is used so it shouldn't be being blocked by anything else outside of the controllable physics serialization checks.